### PR TITLE
Ensure object store plugin processes are cleaned up after restore and after BSL validation during server start up 

### LINKED ISSUE #2039: Plugin processes for object stores not cleared out after restore
**What steps did you take and what happened:**
I ran several Velero restores and I'm observing that object store plugin processes are left behind after the restore completes. Object store plugin processes from backups are cleaned out as expected, however it's on the restore that these stale object store plugin processes are left behind:

Below is the output from running ps cmd on the Velero pod after several backups:
```
$ velero backup create nginx-backup-1 --include-namespaces nginx
$ velero backup create nginx-backup-2 --include-namespaces nginx
nobody@velero-76c6b59b48-6f88l:/$ ps -aux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
nobody       1  0.3  1.2 144916 50088 ?        Ssl  18:32   0:00 /velero server --log-level debug
nobody      32  0.0  0.5 132720 21172 ?        Sl   18:32   0:00 /plugins/velero-plugin-for-aws --log-level debug
nobody     110  0.0  0.0  18504  3392 pts/0    Ss   18:34   0:00 bin/bash
nobody     118  0.0  0.0  34396  2864 pts/0    R+   18:34   0:00 ps -aux
```

I then ran 2 restores and below is the ps output on the Velero pod:
```
$ velero restore create --from-backup nginx-backup-1 --namespace-mappings nginx:nginx-1
$ velero restore create --from-backup nginx-backup-1 --namespace-mappings nginx:nginx-2
$ kubectl exec -it deploy/velero -n velero bin/bash
nobody@velero-76c6b59b48-6f88l:/$ ps -aux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
nobody       1  0.2  1.2 144916 50380 ?        Ssl  18:32   0:00 /velero server --log-level debug
nobody      32  0.0  0.5 132720 21044 ?        Sl   18:32   0:00 /plugins/velero-plugin-for-aws --log-level debug
nobody     154  0.3  0.7 132720 31948 ?        Sl   18:37   0:00 /plugins/velero-plugin-for-aws --log-level debug
nobody     191  0.3  0.5 132720 22504 ?        Sl   18:37   0:00 /plugins/velero-plugin-for-aws --log-level debug
nobody     203  0.0  0.0  18504  3408 pts/0    Ss   18:37   0:00 bin/bash
nobody     211  0.0  0.0  34396  2900 pts/0    R+   18:37   0:00 ps -aux

```

Running 10 restores:
```
$ kubectl exec -it deploy/velero -n velero bin/bash
nobody@velero-76c6b59b48-6f88l:/$ ps -aux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
nobody       1  0.3  1.5 144916 63548 ?        Ssl  18:32   0:01 /velero server --log-level debug
nobody      32  0.0  0.5 132720 21044 ?        Sl   18:32   0:00 /plugins/velero-plugin-for-aws --log-level debug
nobody     154  0.1  0.7 132720 31948 ?        Sl   18:37   0:00 /plugins/velero-plugin-for-aws --log-level debug
nobody     191  0.1  0.5 132720 22504 ?        Sl   18:37   0:00 /plugins/velero-plugin-for-aws --log-level debug
nobody     229  0.2  0.6 132720 27752 ?        Sl   18:37   0:00 /plugins/velero-plugin-for-aws --log-level debug
nobody     256  0.3  0.7 132720 31100 ?        Sl   18:37   0:00 /plugins/velero-plugin-for-aws --log-level debug
nobody     285  0.3  0.8 132720 32656 ?        Sl   18:37   0:00 /plugins/velero-plugin-for-aws --log-level debug
nobody     312  0.4  0.5 132720 21584 ?        Sl   18:37   0:00 /plugins/velero-plugin-for-aws --log-level debug
nobody     339  0.4  0.5 132720 21976 ?        Sl   18:37   0:00 /plugins/velero-plugin-for-aws --log-level debug
nobody     368  0.4  0.6 132720 26180 ?        Sl   18:37   0:00 /plugins/velero-plugin-for-aws --log-level debug
nobody     392  0.6  0.5 132720 22176 ?        Sl   18:38   0:00 /plugins/velero-plugin-for-aws --log-level debug
nobody     418  0.5  0.5 132720 20600 ?        Sl   18:38   0:00 /plugins/velero-plugin-for-aws --log-level debug
nobody     447  0.7  0.5 132720 21588 ?        Sl   18:38   0:00 /plugins/velero-plugin-for-aws --log-level debug
nobody     473  1.0  0.5 132720 21084 ?        Sl   18:38   0:00 /plugins/velero-plugin-for-aws --log-level debug
nobody     495  0.0  0.0  18504  3384 pts/0    Ss   18:38   0:00 bin/bash
nobody     502  0.0  0.0  34396  2920 pts/0    R+   18:38   0:00 ps -aux
```

**What did you expect to happen:**
No stale plugin processes are left post backup/restore.

**The output of the following commands will help us better understand what's going on**:
(Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)

* `kubectl logs deployment/velero -n velero`
https://gist.github.com/betta1/f15e47eb04433b7328fd4e93edeb1f63

* `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
* `velero backup logs <backupname>`
* `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
```
Name:         nginx-backup-1-20191104103701
Namespace:    velero
Labels:       <none>
Annotations:  <none>
Phase:  Completed
Backup:  nginx-backup-1
Namespaces:
 Included:  *
 Excluded:  <none>
Resources:
 Included:        *
 Excluded:        nodes, events, events.events.k8s.io, backups.velero.io, restores.velero.io, resticrepositories.velero.io
 Cluster-scoped:  auto
Namespace mappings:  nginx=nginx-1
Label selector:  <none>
Restore PVs:  auto
```
* `velero restore logs <restorename>`
https://gist.github.com/betta1/e5e6a2f7f9344dff7733747fd6b3ae00

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
This issue seems to occur during restore and on server startup (probably during BSL validation?).

After some time I'm seeing restores failing with "Unrecognized remote plugin message:
This usually means that the plugin is either invalid or simply
needs to be recompiled to support the latest protocol." 
```
velero describe restore nginx-backup-1-20191104194601
Name:         nginx-backup-1-20191104194601
Namespace:    velero
Labels:       <none>
Annotations:  <none>
Phase:  PartiallyFailed (run 'velero restore logs nginx-backup-1-20191104194601' for more information)
Errors:
 Velero:  Unrecognized remote plugin message:
This usually means that the plugin is either invalid or simply
needs to be recompiled to support the latest protocol.
 Cluster:    <none>
 Namespaces: <none>
Backup:  nginx-backup-1
Namespaces:
 Included:  nginx
 Excluded:  <none>
Resources:
 Included:        *
 Excluded:        nodes, events, events.events.k8s.io, backups.velero.io, restores.velero.io, resticrepositories.velero.io
 Cluster-scoped:  auto
Namespace mappings:  nginx=velero-cron-restores-1572896760-c9rb5
Label selector:  <none>
Restore PVs:  auto
```

**Environment:**

- Velero version (use `velero version`): v1.2.0-beta.1
- Velero features (use `velero client config get features`): 
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by betta1**
This fixes #2039

### CODE REVIEW

**Comment by skriss** [comment]
@betta1 this looks good to me - I'll do some testing today as well. Thanks for the PR!

**Comment by skriss** [comment]
FYI, basic testing LGTM - can confirm it fixes the issue, and I didn't encounter any other problems

**Comment by skriss** [comment]
@carlisia @nrb @prydonius PTAL as well, thanks!

File: pkg/controller/restore_controller.go [hunk 1]
```diff
@@ -227,7 +227,7 @@ func (c *restoreController) processRestore(restore *api.Restore) error {
 	// our per-restore log (which is instantiated within c.runValidatedRestore).
 	pluginManager := c.newPluginManager(c.logger)
 	info := c.validateAndComplete(restore, pluginManager)
-	pluginManager.CleanupClients()
+	defer pluginManager.CleanupClients()
```
[the remark this answers is not in the document]
Comment (nrb) [reply pkg/controller/restore_controller.go]: I'm curious, why have a defer here at all? I know you tested this Steve, but I'm unclear as to why the defer is making a difference here. Reading the code at the moment, it appears that the processes should be cleaned up either way, only with this change it happens a little later.

[the remark this answers is not in the document]
Comment (nrb) [reply pkg/controller/restore_controller.go]: :+1: Thanks, that makes sense!

**Pull request merged**